Mage.Sets/src/mage/cards/d/DralnusCrusade.java

```diff
@@ -0,0 +1,126 @@
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.d;
+
+import java.util.UUID;
+
+import mage.ObjectColor;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.common.continuous.BoostAllEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.CardTypePredicate;
+import mage.filter.predicate.mageobject.SubtypePredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author brikr
+ */
+public class DralnusCrusade extends CardImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("Goblin creatures");
+
+    static {
+        filter.add(new SubtypePredicate("Goblin"));
+    }
+
+    public DralnusCrusade(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{B}{R}");
+
+        // Goblin creatures get +1/+1.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new BoostAllEffect(1, 1, Duration.WhileOnBattlefield, filter, false)));
+
+        // All Goblins are black and are Zombies in addition to their other creature types.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new DralnusCrusadeEffect()));
+    }
+
+    public DralnusCrusade(final DralnusCrusade card) {
+        super(card);
+    }
+
+    @Override
+    public DralnusCrusade copy() {
+        return new DralnusCrusade(this);
+    }
+}
+
+class DralnusCrusadeEffect extends ContinuousEffectImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("Goblin creatures");
+
+    static {
+        filter.add(new SubtypePredicate("Goblin"));
+    }
+
+    public DralnusCrusadeEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral);
+        staticText = "All Goblins are black and are Zombies in addition to their other creature types";
+    }
+
+    @Override
+    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
+        for(Permanent permanent : game.getState().getBattlefield().getActivePermanents(filter, source.getControllerId(), source.getSourceId(), game)) {
+            switch (layer) {
+                case TypeChangingEffects_4:
+                    if(!permanent.getSubtype(game).contains("Zombie")) {
```

Comment:
You need to check with prrmanent.hasSubtype. Otherwise the permanents with generic subtypes are missed.